File: lib/qt5/PokemonListComboBox.cpp
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <pkmn/qt5/PokemonListComboBox.hpp>

#include <pkmn/database/lists.hpp>

namespace pkmn { namespace qt5 {

    PokemonListComboBox::PokemonListComboBox(
        int generation,
        bool includePrevious,
        QWidget* parent
    ): QComboBox(parent)
    {
        std::vector<std::string> pokemon = pkmn::database::get_pokemon_list(
                                               generation,
                                               includePrevious
                                           );
        for(auto iter = pokemon.begin(); iter != pokemon.end(); ++iter) {
            addItem(QString::fromStdString(*iter));
        }

        setEditable(false);
    }

}}

// Generated at build time
#include "../include/pkmn/qt5/moc_PokemonListComboBox.cpp"

File: lib/swig/java/ComboBoxes/PokemonListComboBox.java
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.JComboBox;

/**
 * A JComboBox filled with a list of Pokémon in (and up to) a given generation.
 */
public class PokemonListComboBox extends JComboBox<String> {
    public PokemonListComboBox(int generation, boolean includePrevious) {
        StringVector pokemonList = Database.getPokemonList(generation, includePrevious);
        for(long i = 0; i < pokemonList.size(); i++) {
            this.addItem(pokemonList.get((int)i));
        }
        this.setEditable(false);
    }
}
